[8.0][FIX]sale_commission: Only create commissions in customer invoices.

jesusVMayor opened the pull request:
Fixes the creation of account.invoice.line.agent in supplier invoices. It occurs when a customer with agent is also an agent.

jesusVMayor committed:
[FIX]sale_commission,sale_commission_product: Only create and settle commissions in customer invoices.

pedrobaeza commented on sale_commission/models/account_invoice.py:
```diff
@@ -62,26 +62,26 @@ class AccountInvoiceLine(models.Model):
 
     @api.model
-    def _default_agents(self):
+    def create(self, vals):
```
You must not replace the default. It's there for something: modify in screen the default commission values if needed.

ivantodorovich commented on sale_commission_product/models/account_invoice.py:
```diff
@@ -17,21 +17,22 @@ def product_id_change(
-                                   })
-
-                res['value']['agents'] = [(0, 0, x) for x in agent_list]
+        if type == 'out_invoice':
```
what about out_refund?

jesusVMayor commented:
Added out_refund.
I added a check in the settlement wizard because the only thing that prevents commissions creation for the supplier invoices its the partner_id in contexts it is not passed, and this could be changed by any module. I not see another way to improve the control of commissions creation without removing the default.

pedrobaeza commented on sale_commission_product/models/account_invoice.py:
```diff
@@ -17,21 +17,22 @@ def product_id_change(
-                                   })
-
-                res['value']['agents'] = [(0, 0, x) for x in agent_list]
+        if type in ('out_invoice', 'out_refund'):
```
Don't make 2 nested if, but put all conditions in one.

atchuthan reviewed (approved):
minor change as per @pedrobaeza suggestion

Could change label to Needs fixing!!

jesusVMayor commented:
Fixed
